flink-runtime/src/test/java/org/apache/flink/runtime/rpc/resourcemanager/ResourceManagerToTaskExecutorHeartbeatSchedulerTest.java

```diff
@@ -0,0 +1,142 @@
+import org.apache.flink.runtime.rpc.taskexecutor.TaskExecutor;
+import org.apache.flink.runtime.rpc.taskexecutor.TaskExecutorGateway;
+import org.apache.flink.runtime.testutils.ZooKeeperTestUtils;
+import org.apache.flink.runtime.util.ZooKeeperUtils;
+import org.apache.flink.util.TestLogger;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import scala.concurrent.Await;
+import scala.concurrent.Future;
+import scala.concurrent.duration.Deadline;
+import scala.concurrent.duration.FiniteDuration;
+
+import java.io.IOException;
+import java.util.UUID;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class ResourceManagerToTaskExecutorHeartbeatSchedulerTest extends TestLogger {
+
+	@Rule
+	public TemporaryFolder tempFolder = new TemporaryFolder();
+
+	private static TestingServer testingServer;
+
+	private static ActorSystem actorSystem;
+	private static ActorSystem actorSystem2;
+	private static AkkaRpcService akkaRpcService;
+	private static AkkaRpcService akkaRpcService2;
+
+	private static final Timeout timeout = new Timeout(10000, TimeUnit.MILLISECONDS);
+
+	@BeforeClass
+	public static void setup() throws Exception {
+		testingServer = new TestingServer();
+
+		actorSystem = AkkaUtils.createDefaultActorSystem();
+		actorSystem2 = AkkaUtils.createDefaultActorSystem();
+
+		akkaRpcService = new AkkaRpcService(actorSystem, timeout);
+		akkaRpcService2 = new AkkaRpcService(actorSystem2, timeout);
+	}
+
+	@AfterClass
+	public static void teardown() throws Exception {
+		akkaRpcService.stopService();
+		akkaRpcService2.stopService();
+
+		actorSystem.shutdown();
+		actorSystem2.shutdown();
+
+		actorSystem.awaitTermination();
+		actorSystem2.awaitTermination();
+
+		testingServer.stop();
+	}
+
+	@Test
+	public void testStart() throws Exception {
+		ExecutorService executorService = new ForkJoinPool();
+
+		Configuration configuration = ZooKeeperTestUtils
+			.createZooKeeperRecoveryModeConfig(
+				testingServer.getConnectString(),
+				tempFolder.getRoot().getPath());
+
+		CuratorFramework client = ZooKeeperUtils.startCuratorFramework(configuration);
+		LeaderElectionService leaderElectionService = ZooKeeperUtils.createLeaderElectionService(client,
+			configuration);
+		ResourceManager resourceManager = new ResourceManager(akkaRpcService, executorService, leaderElectionService);
+
+		HighAvailabilityServices highAvailabilityServices = new NonHaServices(resourceManager.getAddress());
+		ResourceID resourceID = ResourceID.generate();
+		TaskExecutor taskExecutor = new TaskExecutor(akkaRpcService2, highAvailabilityServices, resourceID);
+
+		resourceManager.start();
+		taskExecutor.start();
```

Comment:
I think that we don't have to start a full-blown cluster with ZooKeeper to test the heartbeat system. It should be enough to mock the receiving ends.